core/src/main/java/ai/timefold/solver/core/impl/move/director/ListVariableAfterUnassignmentAction.java

```diff
@@ -0,0 +1,20 @@
+package ai.timefold.solver.core.impl.move.director;
+
+import ai.timefold.solver.core.api.move.Rebaser;
+import ai.timefold.solver.core.impl.domain.variable.descriptor.ListVariableDescriptor;
+import ai.timefold.solver.core.impl.score.director.VariableDescriptorAwareScoreDirector;
+
+record ListVariableAfterUnassignmentAction<Solution_>(Object element,
+        ListVariableDescriptor<Solution_> variableDescriptor) implements ChangeAction<Solution_> {
+
+    @Override
+    public void undo(VariableDescriptorAwareScoreDirector<Solution_> scoreDirector) {
+        scoreDirector.beforeListVariableElementAssigned(variableDescriptor, element);
```

Comment:
No actions here?

Reply:
An action is a command to undo something which already happened.
This is an after-unassigned action, so the inverse of that is before-assigned.